### PR TITLE
PCHR-3693: Only display workflows when creating cases

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR limits the options available when creating a new case to workflow case types only.

## Before
pr-before
All case types can be selected.

## After
pr-after
Only workflow case types can be selected.

## Technical Details
In order to limit the case types we need to know the id of the custom category field. For this case, we augment the `CustomFieldIds` and add the variable to the dashboard and contacts' task page:

```php
  public function shouldHandle($page) {
    $pageClassName = get_class($page);
    $pagesWhereTheVarsAreDefined = [
      Civi\Angular\Page\Main::class,
      CRM_Tasksassignments_Page_Dashboard::class,
      CRM_Tasksassignments_Page_Tasks::class,
    ];

    return in_array($pageClassName, $pagesWhereTheVarsAreDefined);
  }
```

To pass the custom field ids to Angular the `config` constant was updated:

```js
angular.module('tasks-assignments.constants', []).constant('config', {
  // ...
  customFieldIds: CRM.vars.customFieldIds,
  // ...
});
```

And to filter the actual case types by workflow, the modal assignment controller was updated in the following manner:

```js
    vm.workflowActivityTypes = getWorkflowActivityTypes();

    function getWorkflowActivityTypes () {
      var CATEGORY_FIELD = 'custom_' + config.customFieldIds['caseType.category'];
      var WORKFLOW_TYPE = 'Workflow';

      return $rootScope.cache.assignmentType.arr.filter(function (assignment) {
        return assignment[CATEGORY_FIELD] === WORKFLOW_TYPE;
      });
    }
```
It's important to know that `$rootScope.cache.assignmentType.arr` holds a list of all case types, and we just filter them using the custom category field.

Finally, we filter the options by using the `workflowActivityTypes` property on the view:

```html
<select name="assignment" 
  ... 
  ng-options="value.id as value.title for value in workflowActivityTypes;"> <!-- here -->
  <option value="">- select -</option>
</select>
```
-----

✅Manual Tests - done
✅Jasmine Tests - included and passed
✅JS distributive files - included
⏹CSS distributive files - not needed
⏹Backstop tests - not needed
⏹PHP Unit Tests - not needed

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js [hunk 1]
```diff
@@ -516,7 +541,7 @@ define([
         data: resolvedValues.data,
         defaultAssigneeOptions: resolvedValues.defaultAssigneeOptions,
         session: resolvedValues.session,
-        config: {},
+        config: resolvedValues.config,
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js]: What is the purpose of this change?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js [hunk 2]
```diff
@@ -67,6 +67,30 @@ define([
...
+        expectedWorkflowActivityTypes = _.values(caseTypes).filter(function (caseType) {
+          return caseType[CATEGORY_FIELD] === WORKFLOW_TYPE;
+        });
+
+        initController({
+          config: { customFieldIds: { 'caseType.category': 100 } }
+        });
+      });
+
+      it('stores a list of workflow activity types', function () {
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js]: Please change this to `stores a list of workflow activity types only`. This way we know that its supposed to return only the workflow types.

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Hook/PageRun/CustomFieldIds.php [hunk 3]
```diff
@@ -42,7 +42,14 @@ public function handle($page) {
...
   public function shouldHandle($page) {
-    return $page instanceof Civi\Angular\Page\Main;
+    $pageClassName = get_class($page);
+    $pagesWhereTheVarsAreDefined = [
+      Civi\Angular\Page\Main::class,
+      CRM_Tasksassignments_Page_Dashboard::class,
+      CRM_Tasksassignments_Page_Tasks::class,
+    ];
+
+    return in_array($pageClassName, $pagesWhereTheVarsAreDefined);
```
Comment (ajesamson) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Hook/PageRun/CustomFieldIds.php]: Return type is missing in the comment declaration ☝️

File: uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js (hunk 1 above)
Comment (reneolivo) [reply uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js]: This allows passing a custom `config` object, which is one of the dependencies for the modal assignment. One of the tests takes advantage of this and sets the config object equal to `{ customFieldIds: { 'caseType.category': 100 } }` where 100 is the number of the custom field as defined in the mock data.

https://github.com/compucorp/civihr-tasks-assignments/pull/409/files#diff-8c02655fb75ef8db8e05a681600238d4R85
